internal/app/concurrent_proxy_stage_test.go

```diff
@@ -152,6 +153,34 @@ func (s *ConcurrentProxyStage) the_concurrent_requests_are_sent() {
 	s.assert.NoError(err)
 }
 
+func (s *ConcurrentProxyStage) the_concurrent_requests_are_sent_with_attempt_based_modifier() {
+	err := s.pact.Verify(func() (err error) {
+		attempt := 2
+		s.proxy.ForInteraction(postNamePactWithAnyName).AddModifier("$.status", fmt.Sprintf("%d", http.StatusConflict), &attempt)
```

Comment:
Jjust wondering, is it out of scope to verify adding different modifiers on different attempts within same test?
like on 2nd attempt 409 and body1 on 3rd 418 and body2
not sure there is a use case but would be a valid test to verify interactions are correctly modified, if it is out of scope, feel free to ignore

Reply:
@sofib-form3 We are doing this in a different branch. Will raise a PR soon.